EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -26,10 +26,13 @@ public static class PEP_Analysis
         private static Dictionary<string, Dictionary<int, Tuple<double, double>>> fileSpecificTimeDependantHydrophobicityAverageAndDeviation_modified = new Dictionary<string, Dictionary<int, Tuple<double, double>>>();
         private static Dictionary<string, Dictionary<int, Tuple<double, double>>> fileSpecificTimeDependantHydrophobicityAverageAndDeviation_CZE = new Dictionary<string, Dictionary<int, Tuple<double, double>>>();
 
-        public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch> psms, string searchType, List<(string fileName, CommonParameters fileSpecificParameters)> fileSpecificParameters)
+        public static string ComputePEPValuesForAllPSMsGeneric(List<PeptideSpectralMatch> psms, string searchType, List<(string fileName, CommonParameters fileSpecificParameters)> fileSpecificParameters, string outputFolder)
         {
             string[] trainingVariables = PsmData.trainingInfos[searchType];
 
+            //ensure that the order is always stable.
+            psms.OrderByDescending(p => p.Score).ThenBy(p => p.FdrInfo.QValue).ThenBy(p => p.FullSequence).ThenBy(p => p.ProteinAccession).ToList();
```

Comment:
This isn't doing anything right now.
needs to be "psms = psms.OrderByDescending(...)"
Need a unit test that this is being done.

Reply:
good catch. I changed to psms = psms...

Reply:
changed to psms = psms..... Do I still need a unit test? I think no, but what do i know.